packages/protocol/contracts/layer1/based/TaikoInbox.sol

```diff
@@ -294,6 +294,8 @@ abstract contract TaikoInbox is EssentialContract, ITaikoInbox, IProposeBatch, I
 
                     // Invalidate the conflict transition
                     state.transitions[slot][tid].blockHash = 0;
+                    metasLength = i + 1;
```

Comment:
We also need to change the length of `ctxs`  as `ctxs` is used by `verifyProof`.